libs/checkpoint-postgres/src/index.ts

```diff
@@ -132,19 +132,27 @@ export class PostgresSaver extends BaseCheckpointSaver {
     const SCHEMA_TABLES = getTablesWithSchema(this.options.schema);
     try {
       await client.query(`CREATE SCHEMA IF NOT EXISTS ${this.options.schema}`);
-      let version = -1;
+      let version = 0;
+      const MIGRATIONS = getMigrations(this.options.schema);
+
+      // Run first migration to ensure checkpoint_migrations table exists
+      await client.query(MIGRATIONS[0]);
+
       try {
         const result = await client.query(
           `SELECT v FROM ${SCHEMA_TABLES.checkpoint_migrations} ORDER BY v DESC LIMIT 1`
         );
         if (result.rows.length > 0) {
           version = result.rows[0].v;
         }
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      } catch (error: any) {
+      } catch (error: unknown) {
```

Comment:
The original intention behind this logic was to ensure that we'd only attempt to execute migrations if they needed to be executed, as you'd expect this to be a No-op otherwise. The query in question should be benign, but you can imagine how someone looking at a query log from this might be a bit surprised that it attempts to execute DDL operations when none are necessary.

I suspect that what's actually happening in #903 is that the error checking in this catch block isn't working as intended, probably because the string match is failing in certain cases. We should probably just check for `/relation "[^"]+" does not exist/` or similar.